lib/controllers.js

```diff
@@ -72,20 +65,26 @@ module.exports = function(rootDirectory, logger, indexDocument, errorDocument) {
     logger.error(
       'Object "%s" in bucket "%s" does not exist',
       keyName,
-      req.bucket.name
+      req.params.bucket
     );
 
     if (indexDocument) {
       if (errorDocument) {
-        fileStore.getObject(req.bucket, errorDocument, (err, object, data) => {
-          if (err) {
-            // eslint-disable-next-line no-console
-            console.error("Custom Error Document not found: " + errorDocument);
-            return notFoundResponse(req, res);
-          } else {
-            return buildResponse(req, res, 404, object, data);
+        store.getObject(
+          req.params.bucket,
+          errorDocument,
+          (err, object, data) => {
+            if (err) {
+              // eslint-disable-next-line no-console
+              console.error(
+                "Custom Error Document not found: " + errorDocument
+              );
+              return notFoundResponse(req, res);
+            } else {
+              return buildResponse(req, res, 404, object, data);
```

Comment:
This call to `buildResponse` is still using the old function signature (5 params instead of 4 in your changes).